### PR TITLE
Test Intergration with a console app who use an ApiReactive

### PR DESCRIPTION
**Pull request by fle-bach**
+ circleci migration

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-[![ReactiveXComponent.Net Nuget](https://img.shields.io/nuget/v/ReactiveXComponent.Net.svg)](https://www.nuget.org/packages/ReactiveXComponent.Net)  [![ReactiveXComponent.Net Downloads](https://img.shields.io/nuget/dt/ReactiveXComponent.Net.svg)](https://www.nuget.org/packages/ReactiveXComponent.Net)  [![Build Status](https://travis-ci.org/xcomponent/ReactiveXComponent.Net.svg?branch=master)](https://travis-ci.org/xcomponent/ReactiveXComponent.Net)  [![](http://slack.xcomponent.com/badge.svg)](http://slack.xcomponent.com/)
+[![ReactiveXComponent.Net Nuget](https://img.shields.io/nuget/v/ReactiveXComponent.Net.svg)](https://www.nuget.org/packages/ReactiveXComponent.Net)  [![ReactiveXComponent.Net Downloads](https://img.shields.io/nuget/dt/ReactiveXComponent.Net.svg)](https://www.nuget.org/packages/ReactiveXComponent.Net) [![CircleCI](https://circleci.com/gh/fle-bach/ReactiveXComponent.Net.svg?style=svg)](https://circleci.com/gh/fle-bach/ReactiveXComponent.Net) [![](http://slack.xcomponent.com/badge.svg)](http://slack.xcomponent.com/)
```
Comment (nicolaserny) [inline README.md]: Update circleci badge link to the xcomponent account

File: build.cake [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
+#addin "Cake.FileHelpers&version=3.1.0"
+#addin "Cake.Incubator&version=3.0.0"
+#addin "Cake.DoInDirectory&version=3.2.0"
 #load "cake.scripts/utilities.cake"
 
 var target = Argument("target", "Build");
-var buildConfiguration = Argument("buildConfiguration", "Release");
+var buildConfiguration = Argument("buildConfiguration", "Debug");
+var distribution = Argument("distribution", "Community");
+var buildVersion = Argument("buildVersion", "5.0.0-B1");
```
Comment (nicolaserny) [inline build.cake]: Do you need buildVersion ? (it duplicates the version variable)

File: build.cake [hunk 3]
```diff
@@ -1,18 +1,34 @@
...
-#addin "Cake.Incubator&version=1.6.0"
+#addin "Cake.FileHelpers&version=3.1.0"
+#addin "Cake.Incubator&version=3.0.0"
+#addin "Cake.DoInDirectory&version=3.2.0"
 #load "cake.scripts/utilities.cake"
 
 var target = Argument("target", "Build");
-var buildConfiguration = Argument("buildConfiguration", "Release");
+var buildConfiguration = Argument("buildConfiguration", "Debug");
+var distribution = Argument("distribution", "Community");
```
Comment (nicolaserny) [inline build.cake]: Distribution is useless for this project

File: build.cake [hunk 4]
```diff
@@ -1,18 +1,34 @@
...
+var buildVersion = Argument("buildVersion", "5.0.0-B1");
 var version = Argument("buildVersion", "1.0.0-build1");
+var vsVersion = Argument("vsVersion", "VS2017");
 var apiKey = Argument("nugetKey", "");
 var setAssemblyVersion = Argument<bool>("setAssemblyVersion", false);
 
+var wixVersion = FormatWixVersion(buildVersion);
```
Comment (nicolaserny) [inline build.cake]: wixVersion is useless ?

File: build.cake [hunk 5]
```diff
@@ -24,12 +40,32 @@ Task("Clean")
...
+        var pathHelloWorldIntegrationTest = "./docker/integration_tests/XCProjects/HelloWorldV5/";
+        if (DirectoryExists(pathHelloWorldIntegrationTest + "xcr"))
+        {
+            CleanDirectory(pathHelloWorldIntegrationTest + "xcr");
+        }
+        if (DirectoryExists(pathHelloWorldIntegrationTest + "generated"))
+        {
+            CleanDirectory(pathHelloWorldIntegrationTest + "generated");
+        }
+        if (DirectoryExists(pathHelloWorldIntegrationTest + "rxcAssemblies"))
```
Comment (nicolaserny) [inline build.cake]: misspell rxcAssemblies -> xcrAssemblies

File: build.cake [hunk 6]
```diff
@@ -1,18 +1,34 @@
...
+Setup(context=> {
+    DoInDirectory(@"tools", () => {
+        NuGetInstall("XComponent.Build.Community", new NuGetInstallSettings{ Version=XComponentVersion, ExcludeVersion=true });
+    });
+});
+
 Task("Clean")
     .Does(() =>
     {
+        #break
```
Comment (nicolaserny) [inline build.cake]: Why this comment ?

File: cake.scripts/utilities.cake [hunk 7]
```diff
@@ -169,4 +173,146 @@ var BuildSolution = new Action<string, string, bool, string>((solutionPath, conf
...
+	}
+	else if (buildPart > 60000) 
+	{
+		paddedBuildPart = "9";
+	}
+	return paddedBuildPart;
+}
+
+// convert version (ex : 3.x.y-G15) to wix format (ex : 3.x.y.415)
+public string FormatWixVersion(string buildVersion)
```
Comment (nicolaserny) [inline cake.scripts/utilities.cake]: No need

File: build.cake (hunk 5 above)
Comment (fle-bach) [reply build.cake]: it's 'reactive'xcAssemblies

File: docker/integration_tests/XCProjects/HelloWorldV5/CreateInstancesReactiveApi/CreateInstances/lib/xcassemblies/HelloWorldV5Api.xcApi [hunk 8]
```diff
@@ -0,0 +1,88 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (nicolaserny) [inline docker/integration_tests/XCProjects/HelloWorldV5/CreateInstancesReactiveApi/CreateInstances/lib/xcassemblies/HelloWorldV5Api.xcApi]: Do not push the lib folder

File: docker/integration_tests/XCProjects/HelloWorldV5/Scripts/build.bat [hunk 9]
```diff
@@ -0,0 +1,3 @@
+@rem BUILD BATCH FILE
```
Comment (nicolaserny) [inline docker/integration_tests/XCProjects/HelloWorldV5/Scripts/build.bat]: Scripts folder is generated by xcbuild.
Remove from repository

File: docker/integration_tests/dockerScripts/XCContainer/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,27 @@
+FROM mono:5.12
+USER root
```
Comment (nicolaserny) [inline docker/integration_tests/dockerScripts/XCContainer/Dockerfile]: Optional: bad practice to run your container as root...

File: docker/integration_tests/dockerScripts/XCContainer/wait_for_it.sh [hunk 11]
```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env bash
```
Comment (nicolaserny) [inline docker/integration_tests/dockerScripts/XCContainer/wait_for_it.sh]: Optional: we add wait_for_it twice. We can get it with wget...

File: .circleci/config.yml [hunk 12]
```diff
@@ -0,0 +1,88 @@
+version: 2
+references:
+  restore_repo: &restore_repo
+    restore_cache:
+      name: Restore checked out code
+      keys:
+        - v1-xcomponent-repo-{{ .Branch }}-{{ .Revision }}
```
Comment (maurelio1234) [inline .circleci/config.yml]: rename cache keys?
use something like `vX-reactivexcomponent-net-repo`

File: build.cake [hunk 13]
```diff
@@ -129,4 +159,58 @@ Task("All")
...
+            StartProcess(xcbuild, buildArgs + vsVersion + helloWorldProjectPathParam + GetXCBuildExtraParam());
+        }
+        if (exitCode != 0) {
+            throw new Exception();
+        }
+    });
+
+Task("BuildIntegrationTests")
+  .IsDependentOn("Merge")
+  .IsDependentOn("RestoreNugetPackages")
```
Comment (maurelio1234) [inline build.cake]: why not integrating the restore into the build tasks?

**Comment by maurelio1234** [comment]
In general looks fine for me, added a couple of comments, but they are optional.

Do not forget to squash the PR instead of simply merging

**Pull request merged**